authmate/authmate.go

```diff
@@ -119,13 +126,35 @@ func (a *Agent) checkContainer(ctx context.Context, cid *cid.ID, friendlyName st
 	return cid, nil
 }
 
-func (a *Agent) getCurrentEpoch(ctx context.Context) (uint64, error) {
+func (a *Agent) getEpochDurations(ctx context.Context) (*epochDurations, error) {
 	if conn, _, err := a.pool.Connection(); err != nil {
-		return 0, err
+		return nil, err
 	} else if networkInfo, err := conn.NetworkInfo(ctx); err != nil {
-		return 0, err
+		return nil, err
 	} else {
-		return networkInfo.CurrentEpoch(), nil
+		res := &epochDurations{
+			currentEpoch: networkInfo.CurrentEpoch(),
+			msPerBlock:   networkInfo.MsPerBlock(),
+		}
+
+		networkInfo.NetworkConfig().IterateParameters(func(parameter *netmap.NetworkParameter) bool {
+			if string(parameter.Key()) == "EpochDuration" {
+				data := parameter.Value()
+				for i, j := 0, len(data)-1; i < j; i, j = i+1, j-1 {
+					data[i], data[j] = data[j], data[i]
+				}
+
+				bi := new(big.Int)
+				bi.SetBytes(data)
+				res.blocksInEpoch = bi.Uint64()
```

Comment:
`res.blocksInEpoch = binary.LittleEndian.Uint64(parameter.Value())` should be enough.